tests/main/cgroup-tracking/task.yaml

```diff
@@ -0,0 +1,148 @@
+summary: Each snap app and hook is tracked via cgroups
+systems:
+    # Ubuntu 14.04's special version of systemd doesn't have StartTransientUnit API.
+    # This is covered in more depth in the sister cgroup-tracking-failure test.
+    - -ubuntu-14.04-*
+
+details: |
+    This test creates a snap process that suspends itself and ensures that it
+    placed into the appropriate hierarchy.
+environment:
+    USER/root: root
+    USER/test: test
+prepare: |
+    # This feature depends on the release-app-awareness feature
+    snap set core experimental.refresh-app-awareness=true
+
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-sh
+
+    tests.session -u "$USER" prepare
+
+restore: |
+    snap unset core experimental.refresh-app-awareness
+    tests.session -u "$USER" restore
+    rm -f /tmp/*.pid /tmp/*.stamp
+    if [ "$USER" = root ]; then
+        systemctl --user stop dbus.service || true
+    fi
+
+execute: |
+    # TODO: add a variant that uses a service
+    if [ "$USER" = test ] && "$TESTSTOOLS"/version-compare --strict "$(systemctl --version | head -n 1 | cut -d ' ' -f 2)" -lt 238; then
+        echo "Systemd running as user session manager cannot start transient units"
+        echo "that cross ownership boundary. This test effectively runs through root"
+        echo "that is using ssh to connect to a test machine. Crossing the boundary"
+        echo "of cgroup ownership is allowed by systemd 238 or newer, with a new"
+        echo "dbus API offered by systemd running as system manager, which allows"
```

Comment:
you can probably directly mention org.freedesktop.systemd1.Manager.AttachProcessesToUnit instead of the new dbus API

Reply:
Done